File: python/system-pomoc-do-refaktora/building_builder.py
from model import Building, Floor, Room


class BuildingBuilder(object):

    def __init__(self):
        self._building = None

    def build_new_building(self, name, address):
        self._building = Building(name, address)

    def add_floors(self, first_floor_no, last_floor_no):
        for i in range(first_floor_no, last_floor_no + 1):
            self._building.floors.append(Floor(i))

    def add_rooms(self, floor_no, room_numbers):
        for floor in self._building.floors:
            if floor.number == floor_no:
                for room_number in room_numbers:
                    floor.rooms.append(Room(room_number))

    @property
    def final_building(self):
        return self._building

File: Lab3Solution/bnsit/patterns/model/BuildingBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Lab3Solution.bnsit.patterns.model
{
    class BuildingBuilder
    {
        private Building building;
        internal Building FinalBuilding { get { return building; } }

        internal void BuildNewBuilding(string name, string address) {
            building = new Building(name, address);
        }

        internal void AddEvevations(int first, int last) {
            for (int i = first; i <= last; ++i)
            {
                building.AddElevation(new Elevation(i));
            }
        }

        internal void AddRooms(int elevationNo, List<int> roomNumbers)
        {
            Elevation elevation = building.GetElevation(elevationNo);

            foreach (int roomNo in roomNumbers)
            {
                elevation.AddRoom(new Room(roomNo));
            }
        }
    }
}
